projects/table-builder/src/lib/classes/table-store.ts

```diff
@@ -180,7 +205,52 @@ export class TableStore extends ComponentStore<TableState> {
     if(sorted.length === 0) {
       sorted = this.createPreSort(metaData);
     }
-    return {...state, metaData, sorted};
+    const {order,mds} = this.initializeOrder(state, metaData)
+    return {...state, metaData:mds, sorted, userDefined:{...state.userDefined,order:order}};
   });
 
+  private initializeOrder = (state:TableState,mds: Dictionary<MetaData>) : {order:Dictionary<number>,mds:Dictionary<InternalMetaData>}=> {
```

Comment:
my brain is exploding HELP!

Reply:
### **_BOOOOOOM!!!!!_**

Reply:
can you simplify this code?

Reply:
I tried

Reply:
I think if you just saved all on line 130 it would simplify it significantly.